app/code/Magento/InventoryIndexer/Model/GetStockItemData.php

```diff
@@ -50,6 +51,13 @@ public function execute(string $sku, int $stockId)
             ->from($stockItemTableName, [IndexStructure::QUANTITY, IndexStructure::IS_SALABLE])
             ->where(IndexStructure::SKU . ' = ?', $sku);
 
-        return $connection->fetchRow($select) ?: null;
+        try {
+            return $connection->fetchRow($select) ?: null;
+        } catch (\Exception $e) {
+            throw new LocalizedException(__(
+                'Product with sku "%sku" is not assigned to stock with id "%stock"',
```

Comment:
If we catch `\Exception` (all of possible exceptions) we need to provide more general error message
(BUT NOT Something went wrong)

Reply:
@naydav done.